src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.ts

```diff
@@ -1,53 +1,57 @@
-import { Component, Input, OnChanges } from '@angular/core';
-
-import _ from 'lodash';
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { UsageService } from '../../services/usage.service';
 
 @Component({
   selector: 'cd-usage-bar',
   templateUrl: './usage-bar.component.html',
   styleUrls: ['./usage-bar.component.scss']
 })
-export class UsageBarComponent implements OnChanges {
-  @Input()
-  total: number;
-  @Input()
-  used: any;
-  @Input()
-  warningThreshold?: number;
-  @Input()
-  errorThreshold?: number;
-  @Input()
-  isBinary = true;
-  @Input()
-  decimals = 0;
-  @Input()
-  calculatePerc = true;
-  @Input()
-  title = $localize`usage`;
-  @Input()
-  customLegend?: string;
-  @Input()
-  customLegendValue?: string;
-  @Input()
-  showFreeToolTip = true;
-  @Input()
-  showMultisiteTooltip = false;
+export class UsageBarComponent implements OnInit {
+  @Input() total: number = 0;
+  @Input() used: any = 0;
+  @Input() warningThreshold?: number = 0;
+  @Input() errorThreshold?: number = 0;
+  @Input() isBinary = true;
+  @Input() decimals = 0;
+  @Input() title = $localize`usage`;
 
-  usedPercentage: number;
-  freePercentage: number;
+  @ViewChild('myComponent') myComponent!: ElementRef;
 
-  ngOnChanges() {
-    if (this.calculatePerc) {
-      this.usedPercentage = this.total > 0 ? (this.used / this.total) * 100 : 0;
-      this.freePercentage = 100 - this.usedPercentage;
-    } else {
-      if (this.used) {
-        this.used = this.used.slice(0, -1);
-        this.usedPercentage = Number(this.used);
-        this.freePercentage = 100 - this.usedPercentage;
-      } else {
-        this.usedPercentage = 0;
+  color: string = '#0043ce';
```

Comment:
We shouldn't add color literals to TS logic. This should be handled in the CSS or HTML files.